File: templates/problems/01_arrays_hashing/02_valid_anagram/python/solution.py
def solve_is_anagram(s: str, t: str) -> bool:
    """Check if two strings are anagrams using character counting.

    Time: O(n), Space: O(1) - fixed 26-letter alphabet
    """
    if len(s) != len(t):
        return False
    count = [0] * 26
    for i in range(len(s)):
        count[ord(s[i]) - ord('a')] += 1
        count[ord(t[i]) - ord('a')] -= 1
    for c in count:
        if c != 0:
            return False
    return True

File: templates/problems/01_arrays_hashing/02_valid_anagram/go/solution.go
package valid_anagram

// solveIsAnagram checks if two strings are anagrams using character counting.
// Time: O(n), Space: O(1) — fixed 26-letter alphabet
func solveIsAnagram(s string, t string) bool {
	if len(s) != len(t) {
		return false
	}
	var count [26]int
	for i := 0; i < len(s); i++ {
		count[s[i]-'a']++
		count[t[i]-'a']--
	}
	for _, c := range count {
		if c != 0 {
			return false
		}
	}
	return true
}
